src/averageAges.js

```diff
@@ -15,11 +20,17 @@
  * @return {number}
  */
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  const targetSelector = century === undefined
+    ? person => (person.sex === 'm')
+    : person => (
+      person.sex === 'm'
+      && century === Math.ceil(person.died / 100)
+    );
+
+  const men = people.filter(targetSelector);
+  const menAge = men.map(man => man.died - man.born);
```

Comment:
You don't need to map the array. You can use only `reduce` from `getAverage` function. Tip: `person.died - person.born`.